qt/main.cpp

```diff
@@ -214,17 +214,21 @@ int main(int argc, char * argv[])
     fmt.setSwapBehavior(QSurfaceFormat::DoubleBuffer);
     fmt.setSwapInterval(1);
     fmt.setDepthBufferSize(16);
-#if defined(OMIM_OS_LINUX)
-    // This is a workaround for older distros, which rely on X.org and Mesa,
-    // where somehow the Mesa driver itself doesn't
+
+    // This is a workaround for some systems,
+    // including MacOs and older Linux distros, which rely on X.org and Mesa.
+    // Where somehow the driver itself doesn't
     // make all the otherwise supported GLSL versions available by default
     // and such requests are somehow disregarded at later stages of execution.
     // This setting here will be potentially overwritten and overruled anyway,
     // and only needed to ensure that we have the needed GLSL compiler available
     // later when we need it.
-    if (app.platformName() == QString("xcb"))
+    if (app.platformName() == QString("xcb") ||
```

Comment:
Why explicitly selecting Opengl ES compatibility profile and version for other platforms is bad idea?
```
android
    cocoa is a platform plugin for macOS.
    directfb
    eglfs is a platform plugin for running Qt5 applications on top of EGL and OpenGL ES 2.0 without an actual windowing system (like X11 or Wayland). For more information, see [EGLFS](https://doc.qt.io/qt-6/embedded-linux.html#eglfs).
    ios (also used for tvOS)
    linuxfb writes directly to the framebuffer. For more information, see [LinuxFB](https://doc.qt.io/qt-6/embedded-linux.html#linuxfb).
    minimal is provided as an examples for developers who want to write their own platform plugins. However, you can use the plugin to run GUI applications in environments without a GUI, such as servers.
    minimalegl is an example plugin.
    offscreen
    qnx
    windows
    wayland is a platform plugin for the Wayland display server protocol, used on some Linux desktops and embedded systems.
    xcb is a plugin for the X11 window system, used on some desktop Linux platforms.
```
?